test/unit/fetchers/url_test.rb

```diff
@@ -0,0 +1,150 @@
+# encoding: utf-8
+# author: Dominik Richter
+# author: Christoph Hartmann
+
+require 'helper'
+
+describe Fetchers::Url do
+  let(:fetcher) { Fetchers::Url }
+
+  it 'registers with the fetchers registry' do
+    reg = Inspec::Fetcher.registry
+    _(reg['url']).must_equal fetcher
+  end
+
+  describe 'testing different urls' do
+    let(:mock_file) { MockLoader.profile_path('complete-metadata') }
+    let(:fetcher) {
+      Class.new(Fetchers::Url) do
+        attr_reader :target, :archive
+        def initialize(target, opts)
+          @target = target
+          @archive = File.new(__FILE__)
+        end
+      end
+    }
+
+    it 'handles a http url' do
+      url = 'http://chef.io.tar.gz'
+      res = fetcher.resolve(url)
+      _(res).must_be_kind_of Fetchers::Local
+      _(res.parent.target).must_equal 'http://chef.io.tar.gz'
```

Comment:
Isn't this actually a bug? I understand that intention, but `URI("http://chef.io.tar.gz").path == ""`